napari/utils/colormaps/colormap_utils.py

```diff
@@ -618,7 +618,30 @@ def ensure_colormap(colormap: ValidColormapArg) -> Colormap:
     """
     with AVAILABLE_COLORMAPS_LOCK:
         if isinstance(colormap, str):
-            name = colormap
+            # Is a colormap with this name (default or custom) already available?
+            custom_cmap = AVAILABLE_COLORMAPS.get(
+                colormap,
+                AVAILABLE_COLORMAPS.get(f"custom-{colormap.lower()}", None),
+            )
+            if custom_cmap is None:
+                custom_cmap = _colormap_from_colors(
+                    colormap, name=f"custom-{colormap.lower()}"
+                )
+                if custom_cmap is None:
+                    custom_cmap = vispy_or_mpl_colormap(colormap)
+
+                for cmap_ in AVAILABLE_COLORMAPS.values():
+                    if (
+                        np.array_equal(cmap_.controls, custom_cmap.controls)
+                        and cmap_.colors.shape == custom_cmap.colors.shape
+                        and np.all(cmap_.colors == custom_cmap.colors)
+                        and cmap_.interpolation == custom_cmap.interpolation
+                    ):
+                        custom_cmap = cmap_
+                        break
+
+            name = custom_cmap.name
+            AVAILABLE_COLORMAPS[name] = custom_cmap
             if name not in AVAILABLE_COLORMAPS:
                 cmap = vispy_or_mpl_colormap(
                     name
```

Comment:
this lines looks obsolete, as function is called in line 631

Reply:
Very well spotted, thanks.